content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1389 @@
+rules only changes the current, in-memory configuration. 
+It does not persist between restarts of the iptables service. 
+So, you need to make sure you save the configuration to make the changes permanent.
+
+Saving the configuration is slightly different between RedHat-based and
+Debian-based systems. 
+On a RedHat-based system (Fedora, CentOS, RHEL, etc), issue the following command:
+
+[source]
+----
+sudo iptables-save > /etc/sysconfig/iptables
+----
+
+On a Debian-based system (Debian, Ubuntu, Mint, etc), issue the
+following command:
+
+[source]
+----
+sudo sh -c "iptables-save > /etc/iptables.rules"
+----
+
+The iptables-restore command will need to be executed manually, or your
+system configured to automatically run it on boot, against the
+/etc/iptables.rules file you have created, in order for your iptables
+configuration to be retained across reboots. 
+On Ubuntu fastest way is to install iptables-persistent after configuring iptables - 
+it will automatically create necessery files from current configuration and load
+them on boot.
+
+[source]
+----
+sudo apt-get install iptables-persistent
+----
+
+See https://help.ubuntu.com/community/IptablesHowTo for other Ubuntu
+options. 
+There are many other resources describing this; please consult
+your system's documentation or search on the internet for information
+specific to your flavor of Linux.
+
+If you are unsure at all about what kind of system you have, consult
+that system's documentation on how to update iptables configuration.
+
+==== Alternate Solution
+
+Note that an earlier version of this documentation had different
+iptables commands for forwarding ports. 
+I found that these did not work for me, and so I researched and 
+came up with the information above. 
+I do not believe that DNAT is required, as it is typically used to publish a
+service from an internal network to a publicly accessible IP. 
+But, if the above forwarding commands do not work for you, you might want to
+consider trying the commands below. 
+Keep in mind that the rules to allow incoming traffic are still required in 
+order for these rules to work:
+
+*iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 80 -j DNAT
+--to-destination 127.0.0.1:8080*
+
+The first rule uses -j DNAT target of the PREROUTING chain in NAT to
+specify a destination IP address and port where incoming packets
+requesting a connection to your Jenkins service can be forwarded.
+
+If you have a default policy of DROP in your FORWARD chain, you will
+need to add a rule to forward all incoming traffic.
+
+*iptables -A FORWARD -i eth0 -m state --state NEW -m tcp -p tcp -d
+127.0.0.1 --dport 8080 -j ACCEPT*
+
+For HTTPS you will need to repeat the above steps but specify port 443
+instead of port 80.
+
+==== Using firewalld
+
+Newer Linux distributions (CentOS 7, RHEL 7, etc.) ship with firewalld
+which serves as a front-end for iptables.  
+Configuration thru firewalld is done via the *firewall-cmd* command.  
```

Comment:
Calling CentOS 7 and Red Hat 7 a "newer" Linux distribution is stretching the word "newer" to the breaking point.  Any operating system is not "newer" when it ships command line git 1.8 when many operating systems ship 2.27.
Was newer when the page was written, but not anymore.
```suggestion
Some Linux distributions (CentOS 8, Red hat Enterprise Linux 8, CentOS 7, etc.) ship with firewalld
which serves as a front-end for iptables.
Configuration thru firewalld is done via the *firewall-cmd* command.
```